entproto/internal/todo/ent/proto/entpb/entpb_user_service.go

```diff
@@ -55,17 +56,34 @@ func toProtoUser(e *ent.User) *User {
 	}
 }
 
+// validateUser validates that all fields are encoded properly and are safe to pass
+// to the ent entity builder.
+func validateUser(x *User, checkId bool) error {
+	if err := runtime.ValidateUUID(x.GetCrmId()); err != nil {
+		return err
+	}
+	if err := runtime.ValidateUUID(x.GetAttachment().GetId()); err != nil {
+		return err
+	}
+	return nil
+}
+
 // Create implements UserServiceServer.Create
 func (svc *UserService) Create(ctx context.Context, req *CreateUserRequest) (*User, error) {
 	user := req.GetUser()
+	if err := validateUser(user, false); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid argument: %s", err)
+	}
 	res, err := svc.client.User.Create().
 		SetBanned(bool(user.GetBanned())).
+		SetCrmID(runtime.MustBytesToUUID(user.GetCrmId())).
```

Comment:
Do you prefer to panic here instead of returning an error? Maybe it is worth mentioning this in the doc.

Reply:
It is guaranteed not to panic b/c a program that reaches here passed the validation code. In one of the next versions I will remove this "setter chain" and treat each field separately for supporting per field nullability etc. In that PR it might be easy to make it nicer as you suggest